weed/filer/filer.go

```diff
@@ -313,6 +350,23 @@ func (f *Filer) UpdateEntry(ctx context.Context, oldEntry, entry *Entry) (err er
 	return f.Store.UpdateEntry(ctx, entry)
 }
 
+func (f *Filer) renameEntry(ctx context.Context, oldEntry, newEntry *Entry, newEntryExists bool) error {
+	if oldEntry.IsDirectory() && !newEntry.IsDirectory() {
+		glog.Errorf("existing %s is a directory", oldEntry.FullPath)
+		return fmt.Errorf("existing %s is a directory", oldEntry.FullPath)
+	}
+	if !oldEntry.IsDirectory() && newEntry.IsDirectory() {
+		glog.Errorf("existing %s is a file", oldEntry.FullPath)
+		return fmt.Errorf("existing %s is a file", oldEntry.FullPath)
+	}
+
+	newEntry.Attr.Crtime = oldEntry.Attr.Crtime
+	if newEntryExists {
```

Comment:
oldEntry is not deleted

Reply:
oldEntry is deleted in this function: moveSelfEntry
line 204-227

Reply:
I do not see it.

Reply:
https://github.com/seaweedfs/seaweedfs/blob/3003c9e17e06db92f5a426c191ebaab15a1764a1/weed/server/filer_grpc_server_rename.go#L204